nbopen/nbopen.py

```diff
@@ -41,15 +43,12 @@ def nbopen(filename):
         print("Starting new server")
         # Hack: we want to override these settings if they're in the config file.
         # The application class allows 'command line' config to override config
-        # loaded afterwards from the config file. So by specifying config, we
+        # loaded afterwards from the config file. So by specifying argv, we
         # can use this mechanism.
-        cfg = Config()
-        cfg.NotebookApp.file_to_run = os.path.abspath(filename)
-        cfg.NotebookApp.notebook_dir = nbdir
-        cfg.NotebookApp.open_browser = True
-        notebookapp.launch_new_instance(config=cfg,
-                                        argv=[],  # Avoid it seeing our own argv
-                                        )
+        argv = ["--ServerApp.file_to_run='%s'" % os.path.abspath(filename),
+                "--ServerApp.root_dir='%s'" % nbdir,
```

Comment:
I suspect this formatting could go wrong in corner cases like filenames containing `'` . That's not super important, but it would be good to fix it if we can.

Can we still use the option like in the old code of passing in a `Config` object, or has that been removed?

If that's not an option, does it work to put the values in as separate arguments, e.g. `["--ServerApp.file_to_run", os.path.abspath(filename), ...` ? If that's allowed, that should avoid any need for escaping.

Reply:
You're totally right! I did not manage to get the previous way of passing a Config object to work. As far as I could tell from the code in notebook.app, the `**kwargs` that the `launch_instance` function takes are not used anywhere, so the config keyword argument doesn't do anything anymore (and it did not work on my system).

I had planned to check if passing without string interpolation as a separate list element works and then forgot. I checked now before pushing the change, it works fine (also for filenames containing `'`, which otherwise would have been problematic, as you pointed out).

Reply:
Thanks!

I'm slightly surprised the `config` thing doesn't still work - it has to go through many layers, but I think it should still end up caught by this code:

https://github.com/ipython/traitlets/blob/a5f07181eb6ad8fca5574cfb3b053b0c140b0845/traitlets/config/configurable.py#L81

But if we've got a decent way of specifying the same things in command-line arguments, we don't need to work out precisely what's going on. It may be that one of the many intermediate layers is discarding the config argument, intentionally or otherwise.

Reply:
I can't say I understand how this all fits together, but as far as I can tell, the `launch_instance` class method in https://github.com/jupyter-server/jupyter_server/blob/1cb41168c21c9191dd52103f0ec1209d7e1debd0/jupyter_server/extension/application.py#L585C21-L585C21 does not pass its `kwargs` argument on to `initialize_server` (which then calls `make_serverapp` which in turn calls `instance` from configurables and finally the `__init__`), so it gets lost in that step. Maybe that's a bug and `launch_instance` should actually pass `kwargs` through?

Reply:
That sounds like a decent explanation. It's not clear to me either whether that's intentional or an oversight.